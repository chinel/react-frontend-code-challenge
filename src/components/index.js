export { default as DivisionSummary } from "./DivisionSummary/DivisionSummary";
export { default as Header } from "./Header/Header";
export { default as InfoBox } from "./InfoBox/InfoBox";
export { default as ModuleHistory } from "./ModuleHistory/ModuleHistory";
export { default as TableSummary } from "/TableSummary/TableSummary";
