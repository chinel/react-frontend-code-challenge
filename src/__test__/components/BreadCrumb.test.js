import React from "react";
import { render } from "@testing-library/react";
import useStyles from "../../components/BreadCrumb/BreadCrumbStyles";
import { BreadCrumb } from "../../components";

function checkStyles() {
  let values;

  function MockedComponent() {
    values = useStyles();
    return null;
  }
  render(<MockedComponent />);
  return values;
}

describe("<BreadCrumb/>", () => {
  it("renders the <BreadCrumb with populated data", () => {
    const { container, getByText } = render(<BreadCrumb />);

    expect(getByText("Divisions")).toBeTruthy();
    expect(getByText("Modules")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("validate the return values of useStyles hook", () => {
    const validateStlyes = checkStyles();

    expect(validateStlyes.breadCrumb).toBeTruthy();
  });
});
