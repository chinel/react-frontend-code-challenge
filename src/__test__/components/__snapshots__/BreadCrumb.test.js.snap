// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadCrumb/> renders the <BreadCrumb with populated data 1`] = `
<div
  class="makeStyles-breadCrumb-1"
>
  <nav
    aria-label="breadcrumb"
    class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-2 MuiTypography-colorInherit"
          href="/"
        >
          Divisions
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-breadCrumbSeparator-3 MuiSvgIcon-fontSizeSmall"
          focusable="false"
          style="color: rgb(10, 86, 133);"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
          />
        </svg>
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-2 MuiTypography-colorInherit"
          href="#"
        >
          Modules
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
